node/network/collator-protocol/src/validator_side.rs

```diff
@@ -590,23 +676,50 @@ where
 		..Default::default()
 	};
 
+	let next_inactivity_stream = futures::stream::unfold(
```

Comment:
Why is that a stream and not just a delay that you always [reset](https://docs.rs/futures-timer/3.0.2/futures_timer/struct.Delay.html#method.reset)?

Reply:
Because of `futures::select` it must be `Unpin` which means we use `pin_mut!` which takes away our mutable access.

Reply:
https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=5cc083bfc104416883d6e0f00b16185c should work or?

Reply:
TIL.

But actually I think that this `stream::unfold` is more readable than intermingling "scheduler" and "scheduled" logic.